MekHQ/src/mekhq/campaign/parts/SpacecraftEngine.java

```diff
@@ -124,12 +122,12 @@ public Money getStickerPrice() {
     public boolean isSamePartType(Part part) {
         return part instanceof SpacecraftEngine
                 && getName().equals(part.getName())
-                && getTonnage() == ((SpacecraftEngine)part).getTonnage();
+                && getTonnage() == part.getTonnage();
```

Comment:
Double check that this doesn't have an implementation of getUnitTonnage()

Reply:
Same as above